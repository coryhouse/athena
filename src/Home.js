import React from "react";
import { getCourses, deleteCourse } from "./api/courseApi";

class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      courses: [],
      newCourse: {
        title: "",
        category: ""
      }
    };
  }

  // Runs when the component first mounts. So, runs once.
  componentDidMount() {
    getCourses().then(courses => {
      this.setState({ courses });
    });
  }

  handleDelete(id) {
    // Optimistic delete
    const courses = this.state.courses.filter(course => course.id !== id);
    this.setState({ courses });
    alert("Course deleted");
    deleteCourse(id).then(() => {});
  }

  handleChange = ({ target }) => {
    const newCourse = {
      ...this.state.newCourse,
      [target.name]: target.value
    };
    this.setState({ newCourse });
  };

  render() {
    return (
      <>
        <h1>{this.props.title || "No title :("}</h1>

        <form>
          <div>
            <label htmlFor="title">Title</label>
            <br />
            <input
              type="text"
              id="title"
              name="title"
              onChange={this.handleChange}
              value={this.state.newCourse.title}
            />
          </div>

          <div>
            <label htmlFor="category">Category</label>
            <br />
            <input
              type="text"
              id="category"
              name="category"
              onChange={this.handleChange}
              value={this.state.newCourse.category}
            />
          </div>

          <button type="submit">Save Course</button>
        </form>

        <ul>
          {this.state.courses.map(course => {
            return (
              <li key={course.id}>
                <button onClick={() => this.handleDelete(course.id)}>
                  Delete
                </button>
                {course.title}
              </li>
            );
          })}
        </ul>
      </>
    );
  }
}

export default Home;
