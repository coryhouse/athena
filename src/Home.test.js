import React from "react";
import Home from "./Home";
import { render, cleanup } from "react-testing-library";
import renderer from "react-test-renderer";
import "jest-dom/extend-expect";

it("renders page title when passed a title", () => {
  const tree = renderer.create(<Home title="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders 'no title :(' when not passed a title", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

afterEach(cleanup);

it("should render an h1 with the title provided on props", () => {
  // arrange
  const { getByText, debug } = render(<Home title="Example title" />);

  // act
  // assert
  getByText("Example title");

  debug();
});

it("should pass", () => {
  expect(1).toEqual(1);
});
