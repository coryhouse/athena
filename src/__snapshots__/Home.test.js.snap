// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 'no title :(' when not passed a title 1`] = `
Array [
  <h1>
    No title :(
  </h1>,
  <form>
    <div>
      <label
        htmlFor="title"
      >
        Title
      </label>
      <br />
      <input
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="category"
      >
        Category
      </label>
      <br />
      <input
        id="category"
        name="category"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Save Course
    </button>
  </form>,
  <ul />,
]
`;

exports[`renders page title when passed a title 1`] = `
Array [
  <h1>
    test
  </h1>,
  <form>
    <div>
      <label
        htmlFor="title"
      >
        Title
      </label>
      <br />
      <input
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        htmlFor="category"
      >
        Category
      </label>
      <br />
      <input
        id="category"
        name="category"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Save Course
    </button>
  </form>,
  <ul />,
]
`;
